bin/coverage_doctest.py does not work

```pytb
$./bin/coverage_doctest.py
DOCTEST COVERAGE for /Users/aaronmeurer/Documents/Python/sympy/sympy/sympy
======================================================================

Traceback (most recent call last):
  File "./bin/coverage_doctest.py", line 601, in <module>
    no_color=args.no_color, sphinx=args.sphinx)
  File "./bin/coverage_doctest.py", line 541, in go
    verbose=verbose, no_color=no_color, exact=False, sphinx=sphinx)
  File "./bin/coverage_doctest.py", line 549, in go
    any(name in file for name in skip_paths))):
NameError: name 'skip_paths' is not defined
```

It works if you remove the skip_paths check, but we should look into the history of the file to see what that was, because clearly something broke at some point.

I guess I forgot to mention that I had already run the script. My trace was the same as the previous user's, but I'll post it here:
```

======================================================================

sympy.abc: Doctests: 100% (0 of 0)
Traceback (most recent call last):
  File "./coverage_doctest.py", line 623, in <module>
    doctests, total_sphinx, num_functions = go(sympy_top, file, verbose=args.verbose,
  File "./coverage_doctest.py", line 562, in go
    _doctests, _total_sphinx,  _num_functions = go(sympy_top, '%s/%s' % (file, F),
  File "./coverage_doctest.py", line 562, in go
    _doctests, _total_sphinx,  _num_functions = go(sympy_top, '%s/%s' % (file, F),
  File "./coverage_doctest.py", line 579, in go
    return coverage(get_mod_name(file, sympy_top), verbose=verbose,
  File "./coverage_doctest.py", line 509, in coverage
    f_dt, f = process_function(f_name, member, obj,
  File "./coverage_doctest.py", line 311, in process_function
    full_name = _get_arg_list(name, obj)
  File "./coverage_doctest.py", line 231, in _get_arg_list
    if argspec.keywords:
AttributeError: 'FullArgSpec' object has no attribute 'keywords'

```
This doesn't _look_ to me like it's about being written for an older version of Python, unless the name of `keywords` has changed in recent python versions.

With the new code I proposed the output looks like this:
```
======================================================================

sympy.abc: Doctests: 100% (0 of 0)
sympy.algebras.quaternion: Doctests: 81% (13 of 16)
sympy.assumptions.ask: Doctests: 25% (1 of 4)
sympy.assumptions.ask_generated: Doctests: 0% (0 of 2)
Traceback (most recent call last):
  File "./coverage_doctest.py", line 624, in <module>
    doctests, total_sphinx, num_functions = go(sympy_top, file, verbose=args.verbose,
  File "./coverage_doctest.py", line 563, in go
    _doctests, _total_sphinx,  _num_functions = go(sympy_top, '%s/%s' % (file, F),
  File "./coverage_doctest.py", line 563, in go
    _doctests, _total_sphinx,  _num_functions = go(sympy_top, '%s/%s' % (file, F),
  File "./coverage_doctest.py", line 580, in go
    return coverage(get_mod_name(file, sympy_top), verbose=verbose,
  File "./coverage_doctest.py", line 480, in coverage
    c_dt, c, source = process_class(member, obj, c_skipped, c_md,
  File "./coverage_doctest.py", line 400, in process_class
    assert None, type(doc)
AssertionError: <class 'property'>
```

After looking into it, it seems like the `AssertionEror` is caused by `PredicateMeta` lacking an `obj.__doc__`.

Right. As I mentioned before, it looks like `keywords` isn't in use anymore. Looking closer, it seems like it used to be the name of the `**` parameter, which is now named `varkw`. That means that the updated code should be this:
```
    if argspec.varkw:
        arg_list.append(argspec.varkw)
```
The output produced by this code is as follows:
```
======================================================================

sympy.abc: Doctests: 100% (0 of 0)
sympy.algebras.quaternion: Doctests: 81% (13 of 16)
sympy.assumptions.ask: Doctests: 25% (1 of 4)
sympy.assumptions.ask_generated: Doctests: 0% (0 of 2)
Traceback (most recent call last):
  File "./coverage_doctest.py", line 624, in <module>
    doctests, total_sphinx, num_functions = go(sympy_top, file, verbose=args.verbose,
  File "./coverage_doctest.py", line 563, in go
    _doctests, _total_sphinx,  _num_functions = go(sympy_top, '%s/%s' % (file, F),
  File "./coverage_doctest.py", line 563, in go
    _doctests, _total_sphinx,  _num_functions = go(sympy_top, '%s/%s' % (file, F),
  File "./coverage_doctest.py", line 580, in go
    return coverage(get_mod_name(file, sympy_top), verbose=verbose,
  File "./coverage_doctest.py", line 480, in coverage
    c_dt, c, source = process_class(member, obj, c_skipped, c_md,
  File "./coverage_doctest.py", line 400, in process_class
    assert None, type(doc)
AssertionError: <class 'property'>
```

--- a/bin/coverage_doctest.py
+++ b/bin/coverage_doctest.py
@@ -228,7 +228,6 @@ def _get_arg_list(name, fobj):
     # Add var args
     if argspec.varargs:
         arg_list.append(argspec.varargs)
-    #if hasattr(argspec, 'keywords') and argspec.keywords:
     if argspec.varkw:
         arg_list.append(argspec.varkw)
 

--- a/bin/coverage_doctest.py
+++ b/bin/coverage_doctest.py
@@ -72,8 +72,8 @@ def print_header(name, underline=None, color=None):
         print(underline*len(name))
 
 
-def print_coverage(module_path, c, c_md, c_mdt, c_idt, c_sph, f, f_md, f_mdt,
-                   f_idt, f_sph, score, total_doctests, total_members,
+def print_coverage(module_path, c, c_missing_doc, c_missing_doctest, c_indierect_doctest, c_sph, f, f_missing_doc, f_missing_doctest,
+                   f_indierect_doctest, f_sph, score, total_doctests, total_members,
                    sphinx_score, total_sphinx, verbose=False, no_color=False,
                    sphinx=True):
     """ Prints details (depending on verbose) of a module """
@@ -137,19 +137,19 @@ def print_coverage(module_path, c, c_md, c_mdt, c_idt, c_sph, f, f_md, f_mdt,
             print_header('No classes found!')
 
         else:
-            if c_md:
+            if c_missing_doc:
                 print_header('Missing docstrings', '-', not no_color and small_header_color)
-                for md in c_md:
+                for md in c_missing_doc:
                     print('  * ' + md)
-            if c_mdt:
+            if c_missing_doctest:
                 print_header('Missing doctests', '-', not no_color and small_header_color)
-                for md in c_mdt:
+                for md in c_missing_doctest:
                     print('  * ' + md)
-            if c_idt:
+            if c_indierect_doctest:
                 # Use "# indirect doctest" in the docstring to
                 # suppress this warning.
                 print_header('Indirect doctests', '-', not no_color and small_header_color)
-                for md in c_idt:
+                for md in c_indierect_doctest:
                     print('  * ' + md)
                 print('\n    Use \"# indirect doctest\" in the docstring to suppress this warning')
             if c_sph:
@@ -161,17 +161,17 @@ def print_coverage(module_path, c, c_md, c_mdt, c_idt, c_sph, f, f_md, f_mdt,
         if not f:
             print_header('No functions found!')
         else:
-            if f_md:
+            if f_missing_doc:
                 print_header('Missing docstrings', '-', not no_color and small_header_color)
-                for md in f_md:
+                for md in f_missing_doc:
                     print('  * ' + md)
-            if f_mdt:
+            if f_missing_doctest:
                 print_header('Missing doctests', '-', not no_color and small_header_color)
-                for md in f_mdt:
+                for md in f_missing_doctest:
                     print('  * ' + md)
-            if f_idt:
+            if f_indierect_doctest:
                 print_header('Indirect doctests', '-', not no_color and small_header_color)
-                for md in f_idt:
+                for md in f_indierect_doctest:
                     print('  * ' + md)
                 print('\n    Use \"# indirect doctest\" in the docstring to suppress this warning')
             if f_sph:
@@ -284,20 +284,20 @@ def find_sphinx(name, mod_path, found={}):
     found[mod_path] = p.is_imported
     return name in p.is_imported
 
-def process_function(name, c_name, b_obj, mod_path, f_sk, f_md, f_mdt, f_idt,
-                     f_has_doctest, sk_list, sph, sphinx=True):
+def process_function(name, c_name, b_obj, mod_path, f_skip, f_missing_doc, f_missing_doctest, f_indierect_doctest,
+                     f_has_doctest, skip_list, sph, sphinx=True):
     """
     Processes a function to get information regarding documentation.
     It is assume that the function calling this subrouting has already
     verified that it is a valid module function.
     """
-    if name in sk_list:
+    if name in skip_list:
         return False, False
 
     # We add in the end, as inspect.getsourcelines is slow
-    add_md = False
-    add_mdt = False
-    add_idt = False
+    add_missing_doc = False
+    add_missing_doctest = False
+    add_indierect_doctest = False
     in_sphinx = True
     f_doctest = False
     function = False
@@ -312,30 +312,30 @@ def process_function(name, c_name, b_obj, mod_path, f_sk, f_md, f_mdt, f_idt,
     full_name = _get_arg_list(name, obj)
 
     if name.startswith('_'):
-        f_sk.append(full_name)
+        f_skip.append(full_name)
     else:
         doc = obj.__doc__
-        if type(doc) is str:
+        if isinstance(doc, str):
             if not doc:
-                add_md = True
+                add_missing_doc = True
             elif not '>>>' in doc:
-                add_mdt = True
+                add_missing_doctest = True
             elif _is_indirect(name, doc):
-                add_idt = True
+                add_indierect_doctest = True
             else:
                 f_doctest = True
         elif doc is None:
             # this was a function defined in the docstring
             f_doctest = True
         else:
-            raise TypeError('Docstring must be a string or property type')
+            raise TypeError('Current doc type for ', print(obj), ' is ', type(doc), '. Docstring must be a string, property, or none')
 
         function = True
 
         if sphinx:
             in_sphinx = find_sphinx(obj_name, mod_path)
 
-    if add_md or add_mdt or add_idt or not in_sphinx:
+    if add_missing_doc or add_missing_doctest or add_indierect_doctest or not in_sphinx:
         try:
             line_no = inspect.getsourcelines(obj)[1]
         except IOError:
@@ -344,19 +344,19 @@ def process_function(name, c_name, b_obj, mod_path, f_sk, f_md, f_mdt, f_idt,
             return False, False
 
         full_name = "LINE %d: %s" % (line_no, full_name)
-        if add_md:
-            f_md.append(full_name)
-        elif add_mdt:
-            f_mdt.append(full_name)
-        elif add_idt:
-            f_idt.append(full_name)
+        if add_missing_doc:
+            f_missing_doc.append(full_name)
+        elif add_missing_doctest:
+            f_missing_doctest.append(full_name)
+        elif add_indierect_doctest:
+            f_indierect_doctest.append(full_name)
         if not in_sphinx:
             sph.append(full_name)
 
     return f_doctest, function
 
 
-def process_class(c_name, obj, c_skip, c_md, c_mdt, c_idt, c_has_doctest,
+def process_class(c_name, obj, c_skip, c_missing_doc, c_missing_doctest, c_indierect_doctest, c_has_doctest,
                   mod_path, sph, sphinx=True):
     """
     Extracts information about the class regarding documentation.
@@ -384,11 +384,11 @@ def process_class(c_name, obj, c_skip, c_md, c_mdt, c_idt, c_has_doctest,
     doc = obj.__doc__
     if isinstance(doc, str):
         if not doc:
-            c_md.append(full_name)
+            c_missing_doc.append(full_name)
         elif not '>>>' in doc:
-            c_mdt.append(full_name)
+            c_missing_doctest.append(full_name)
         elif _is_indirect(c_name, doc):
-            c_idt.append(full_name)
+            c_indierect_doctest.append(full_name)
         else:
             c_dt = True
             c_has_doctest.append(full_name)
@@ -401,9 +401,7 @@ def process_class(c_name, obj, c_skip, c_md, c_mdt, c_idt, c_has_doctest,
         c_skip.append(c_name)
         return False, False, None
     else:
-        print(obj)
-        print(type(doc))
-        raise TypeError('Docstring must be a string or property type')
+        raise TypeError('Current doc type of ', print(obj), ' is ', type(doc), '. Docstring must be a string, property , or none')
 
     in_sphinx = False
     if sphinx:
@@ -432,19 +430,19 @@ def coverage(module_path, verbose=False, no_color=False, sphinx=True):
         return 0, 0, 0
 
     c_skipped = []
-    c_md = []
-    c_mdt = []
+    c_missing_doc = []
+    c_missing_doctest = []
     c_has_doctest = []
-    c_idt = []
+    c_indierect_doctest = []
     classes = 0
     c_doctests = 0
     c_sph = []
 
     f_skipped = []
-    f_md = []
-    f_mdt = []
+    f_missing_doc = []
+    f_missing_doctest = []
     f_has_doctest = []
-    f_idt = []
+    f_indierect_doctest = []
     functions = 0
     f_doctests = 0
     f_sph = []
@@ -472,7 +470,7 @@ def coverage(module_path, verbose=False, no_color=False, sphinx=True):
         if inspect.isfunction(obj) or inspect.ismethod(obj):
 
             f_dt, f = process_function(member, '', obj, module_path,
-                f_skipped, f_md, f_mdt, f_idt, f_has_doctest, skip_members,
+                f_skipped, f_missing_doc, f_missing_doctest, f_indierect_doctest, f_has_doctest, skip_members,
                 f_sph, sphinx=sphinx)
             if f:
                 functions += 1
@@ -483,8 +481,8 @@ def coverage(module_path, verbose=False, no_color=False, sphinx=True):
         elif inspect.isclass(obj):
 
             # Process the class first
-            c_dt, c, source = process_class(member, obj, c_skipped, c_md,
-                c_mdt, c_idt, c_has_doctest, module_path, c_sph, sphinx=sphinx)
+            c_dt, c, source = process_class(member, obj, c_skipped, c_missing_doc,
+                c_missing_doctest, c_indierect_doctest, c_has_doctest, module_path, c_sph, sphinx=sphinx)
             if not c:
                 continue
             else:
@@ -514,7 +512,7 @@ def coverage(module_path, verbose=False, no_color=False, sphinx=True):
                 if inspect.isfunction(f_obj) or inspect.ismethod(f_obj):
 
                     f_dt, f = process_function(f_name, member, obj,
-                        module_path, f_skipped, f_md, f_mdt, f_idt, f_has_doctest,
+                        module_path, f_skipped, f_missing_doc, f_missing_doctest, f_indierect_doctest, f_has_doctest,
                         skip_members, f_sph, sphinx=sphinx)
                     if f:
                         functions += 1
@@ -542,16 +540,16 @@ def coverage(module_path, verbose=False, no_color=False, sphinx=True):
         sphinx_score = 0
 
     # Sort functions/classes by line number
-    c_md = sorted(c_md, key=lambda x: int(x.split()[1][:-1]))
-    c_mdt = sorted(c_mdt, key=lambda x: int(x.split()[1][:-1]))
-    c_idt = sorted(c_idt, key=lambda x: int(x.split()[1][:-1]))
+    c_missing_doc = sorted(c_missing_doc, key=lambda x: int(x.split()[1][:-1]))
+    c_missing_doctest = sorted(c_missing_doctest, key=lambda x: int(x.split()[1][:-1]))
+    c_indierect_doctest = sorted(c_indierect_doctest, key=lambda x: int(x.split()[1][:-1]))
 
-    f_md = sorted(f_md, key=lambda x: int(x.split()[1][:-1]))
-    f_mdt = sorted(f_mdt, key=lambda x: int(x.split()[1][:-1]))
-    f_idt = sorted(f_idt, key=lambda x: int(x.split()[1][:-1]))
+    f_missing_doc = sorted(f_missing_doc, key=lambda x: int(x.split()[1][:-1]))
+    f_missing_doctest = sorted(f_missing_doctest, key=lambda x: int(x.split()[1][:-1]))
+    f_indierect_doctest = sorted(f_indierect_doctest, key=lambda x: int(x.split()[1][:-1]))
 
-    print_coverage(module_path, classes, c_md, c_mdt, c_idt, c_sph, functions, f_md,
-                   f_mdt, f_idt, f_sph, score, total_doctests, total_members,
+    print_coverage(module_path, classes, c_missing_doc, c_missing_doctest, c_indierect_doctest, c_sph, functions, f_missing_doc,
+                   f_missing_doctest, f_indierect_doctest, f_sph, score, total_doctests, total_members,
                    sphinx_score, total_sphinx, verbose=verbose,
                    no_color=no_color, sphinx=sphinx)
 

--- a/bin/coverage_doctest.py
+++ b/bin/coverage_doctest.py
@@ -72,8 +72,8 @@ def print_header(name, underline=None, color=None):
         print(underline*len(name))
 
 
-def print_coverage(module_path, c, c_missing_doc, c_missing_doctest, c_indierect_doctest, c_sph, f, f_missing_doc, f_missing_doctest,
-                   f_indierect_doctest, f_sph, score, total_doctests, total_members,
+def print_coverage(module_path, c, c_missing_doc, c_missing_doctest, c_indirect_doctest, c_sph, f, f_missing_doc, f_missing_doctest,
+                   f_indirect_doctest, f_sph, score, total_doctests, total_members,
                    sphinx_score, total_sphinx, verbose=False, no_color=False,
                    sphinx=True):
     """ Prints details (depending on verbose) of a module """
@@ -145,11 +145,11 @@ def print_coverage(module_path, c, c_missing_doc, c_missing_doctest, c_indierect
                 print_header('Missing doctests', '-', not no_color and small_header_color)
                 for md in c_missing_doctest:
                     print('  * ' + md)
-            if c_indierect_doctest:
+            if c_indirect_doctest:
                 # Use "# indirect doctest" in the docstring to
                 # suppress this warning.
                 print_header('Indirect doctests', '-', not no_color and small_header_color)
-                for md in c_indierect_doctest:
+                for md in c_indirect_doctest:
                     print('  * ' + md)
                 print('\n    Use \"# indirect doctest\" in the docstring to suppress this warning')
             if c_sph:
@@ -169,9 +169,9 @@ def print_coverage(module_path, c, c_missing_doc, c_missing_doctest, c_indierect
                 print_header('Missing doctests', '-', not no_color and small_header_color)
                 for md in f_missing_doctest:
                     print('  * ' + md)
-            if f_indierect_doctest:
+            if f_indirect_doctest:
                 print_header('Indirect doctests', '-', not no_color and small_header_color)
-                for md in f_indierect_doctest:
+                for md in f_indirect_doctest:
                     print('  * ' + md)
                 print('\n    Use \"# indirect doctest\" in the docstring to suppress this warning')
             if f_sph:
@@ -283,7 +283,7 @@ def find_sphinx(name, mod_path, found={}):
     found[mod_path] = p.is_imported
     return name in p.is_imported
 
-def process_function(name, c_name, b_obj, mod_path, f_skip, f_missing_doc, f_missing_doctest, f_indierect_doctest,
+def process_function(name, c_name, b_obj, mod_path, f_skip, f_missing_doc, f_missing_doctest, f_indirect_doctest,
                      f_has_doctest, skip_list, sph, sphinx=True):
     """
     Processes a function to get information regarding documentation.
@@ -296,7 +296,7 @@ def process_function(name, c_name, b_obj, mod_path, f_skip, f_missing_doc, f_mis
     # We add in the end, as inspect.getsourcelines is slow
     add_missing_doc = False
     add_missing_doctest = False
-    add_indierect_doctest = False
+    add_indirect_doctest = False
     in_sphinx = True
     f_doctest = False
     function = False
@@ -320,7 +320,7 @@ def process_function(name, c_name, b_obj, mod_path, f_skip, f_missing_doc, f_mis
             elif not '>>>' in doc:
                 add_missing_doctest = True
             elif _is_indirect(name, doc):
-                add_indierect_doctest = True
+                add_indirect_doctest = True
             else:
                 f_doctest = True
         elif doc is None:
@@ -334,7 +334,7 @@ def process_function(name, c_name, b_obj, mod_path, f_skip, f_missing_doc, f_mis
         if sphinx:
             in_sphinx = find_sphinx(obj_name, mod_path)
 
-    if add_missing_doc or add_missing_doctest or add_indierect_doctest or not in_sphinx:
+    if add_missing_doc or add_missing_doctest or add_indirect_doctest or not in_sphinx:
         try:
             line_no = inspect.getsourcelines(obj)[1]
         except IOError:
@@ -347,15 +347,15 @@ def process_function(name, c_name, b_obj, mod_path, f_skip, f_missing_doc, f_mis
             f_missing_doc.append(full_name)
         elif add_missing_doctest:
             f_missing_doctest.append(full_name)
-        elif add_indierect_doctest:
-            f_indierect_doctest.append(full_name)
+        elif add_indirect_doctest:
+            f_indirect_doctest.append(full_name)
         if not in_sphinx:
             sph.append(full_name)
 
     return f_doctest, function
 
 
-def process_class(c_name, obj, c_skip, c_missing_doc, c_missing_doctest, c_indierect_doctest, c_has_doctest,
+def process_class(c_name, obj, c_skip, c_missing_doc, c_missing_doctest, c_indirect_doctest, c_has_doctest,
                   mod_path, sph, sphinx=True):
     """
     Extracts information about the class regarding documentation.
@@ -387,7 +387,7 @@ def process_class(c_name, obj, c_skip, c_missing_doc, c_missing_doctest, c_indie
         elif not '>>>' in doc:
             c_missing_doctest.append(full_name)
         elif _is_indirect(c_name, doc):
-            c_indierect_doctest.append(full_name)
+            c_indirect_doctest.append(full_name)
         else:
             c_dt = True
             c_has_doctest.append(full_name)
@@ -432,7 +432,7 @@ def coverage(module_path, verbose=False, no_color=False, sphinx=True):
     c_missing_doc = []
     c_missing_doctest = []
     c_has_doctest = []
-    c_indierect_doctest = []
+    c_indirect_doctest = []
     classes = 0
     c_doctests = 0
     c_sph = []
@@ -441,7 +441,7 @@ def coverage(module_path, verbose=False, no_color=False, sphinx=True):
     f_missing_doc = []
     f_missing_doctest = []
     f_has_doctest = []
-    f_indierect_doctest = []
+    f_indirect_doctest = []
     functions = 0
     f_doctests = 0
     f_sph = []
@@ -469,7 +469,7 @@ def coverage(module_path, verbose=False, no_color=False, sphinx=True):
         if inspect.isfunction(obj) or inspect.ismethod(obj):
 
             f_dt, f = process_function(member, '', obj, module_path,
-                f_skipped, f_missing_doc, f_missing_doctest, f_indierect_doctest, f_has_doctest, skip_members,
+                f_skipped, f_missing_doc, f_missing_doctest, f_indirect_doctest, f_has_doctest, skip_members,
                 f_sph, sphinx=sphinx)
             if f:
                 functions += 1
@@ -481,7 +481,7 @@ def coverage(module_path, verbose=False, no_color=False, sphinx=True):
 
             # Process the class first
             c_dt, c, source = process_class(member, obj, c_skipped, c_missing_doc,
-                c_missing_doctest, c_indierect_doctest, c_has_doctest, module_path, c_sph, sphinx=sphinx)
+                c_missing_doctest, c_indirect_doctest, c_has_doctest, module_path, c_sph, sphinx=sphinx)
             if not c:
                 continue
             else:
@@ -511,7 +511,7 @@ def coverage(module_path, verbose=False, no_color=False, sphinx=True):
                 if inspect.isfunction(f_obj) or inspect.ismethod(f_obj):
 
                     f_dt, f = process_function(f_name, member, obj,
-                        module_path, f_skipped, f_missing_doc, f_missing_doctest, f_indierect_doctest, f_has_doctest,
+                        module_path, f_skipped, f_missing_doc, f_missing_doctest, f_indirect_doctest, f_has_doctest,
                         skip_members, f_sph, sphinx=sphinx)
                     if f:
                         functions += 1
@@ -541,14 +541,14 @@ def coverage(module_path, verbose=False, no_color=False, sphinx=True):
     # Sort functions/classes by line number
     c_missing_doc = sorted(c_missing_doc, key=lambda x: int(x.split()[1][:-1]))
     c_missing_doctest = sorted(c_missing_doctest, key=lambda x: int(x.split()[1][:-1]))
-    c_indierect_doctest = sorted(c_indierect_doctest, key=lambda x: int(x.split()[1][:-1]))
+    c_indirect_doctest = sorted(c_indirect_doctest, key=lambda x: int(x.split()[1][:-1]))
 
     f_missing_doc = sorted(f_missing_doc, key=lambda x: int(x.split()[1][:-1]))
     f_missing_doctest = sorted(f_missing_doctest, key=lambda x: int(x.split()[1][:-1]))
-    f_indierect_doctest = sorted(f_indierect_doctest, key=lambda x: int(x.split()[1][:-1]))
+    f_indirect_doctest = sorted(f_indirect_doctest, key=lambda x: int(x.split()[1][:-1]))
 
-    print_coverage(module_path, classes, c_missing_doc, c_missing_doctest, c_indierect_doctest, c_sph, functions, f_missing_doc,
-                   f_missing_doctest, f_indierect_doctest, f_sph, score, total_doctests, total_members,
+    print_coverage(module_path, classes, c_missing_doc, c_missing_doctest, c_indirect_doctest, c_sph, functions, f_missing_doc,
+                   f_missing_doctest, f_indirect_doctest, f_sph, score, total_doctests, total_members,
                    sphinx_score, total_sphinx, verbose=verbose,
                    no_color=no_color, sphinx=sphinx)
 

--- a/bin/coverage_doctest.py
+++ b/bin/coverage_doctest.py
@@ -228,8 +228,9 @@ def _get_arg_list(name, fobj):
     # Add var args
     if argspec.varargs:
         arg_list.append(argspec.varargs)
-    if argspec.keywords:
-        arg_list.append(argspec.keywords)
+    #if hasattr(argspec, 'keywords') and argspec.keywords:
+    if argspec.varkw:
+        arg_list.append(argspec.varkw)
 
     # Truncate long arguments
     arg_list = [x[:trunc] for x in arg_list]
@@ -327,7 +328,7 @@ def process_function(name, c_name, b_obj, mod_path, f_sk, f_md, f_mdt, f_idt,
             # this was a function defined in the docstring
             f_doctest = True
         else:
-            assert None, type(doc)
+            raise TypeError('Docstring must be a string or property type')
 
         function = True
 
@@ -355,7 +356,7 @@ def process_function(name, c_name, b_obj, mod_path, f_sk, f_md, f_mdt, f_idt,
     return f_doctest, function
 
 
-def process_class(c_name, obj, c_sk, c_md, c_mdt, c_idt, c_has_doctest,
+def process_class(c_name, obj, c_skip, c_md, c_mdt, c_idt, c_has_doctest,
                   mod_path, sph, sphinx=True):
     """
     Extracts information about the class regarding documentation.
@@ -365,7 +366,7 @@ def process_class(c_name, obj, c_sk, c_md, c_mdt, c_idt, c_has_doctest,
 
     # Skip class case
     if c_name.startswith('_'):
-        c_sk.append(c_name)
+        c_skip.append(c_name)
         return False, False, None
 
     c = False
@@ -381,7 +382,7 @@ def process_class(c_name, obj, c_sk, c_md, c_mdt, c_idt, c_has_doctest,
     c = True
     full_name = "LINE %d: %s" % (line_no, c_name)
     doc = obj.__doc__
-    if type(doc) is str:
+    if isinstance(doc, str):
         if not doc:
             c_md.append(full_name)
         elif not '>>>' in doc:
@@ -395,8 +396,14 @@ def process_class(c_name, obj, c_sk, c_md, c_mdt, c_idt, c_has_doctest,
         # this was a class defined in the docstring
         c_dt = True
         c_has_doctest.append(full_name)
+    elif isinstance(doc, property):
+        # skip class with dynamic doc
+        c_skip.append(c_name)
+        return False, False, None
     else:
-        assert None, type(doc)
+        print(obj)
+        print(type(doc))
+        raise TypeError('Docstring must be a string or property type')
 
     in_sphinx = False
     if sphinx:

